x-pack/plugins/ml/public/application/components/create_data_view_button/create_data_view_button.tsx

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { EuiButton } from '@elastic/eui';
+import { FormattedMessage } from '@kbn/i18n-react';
+import React, { useCallback, useRef } from 'react';
+import { useMlKibana } from '../../contexts/kibana';
+
+export const CreateDataViewButton = ({
+  onDataViewCreated,
+  allowAdHocDataView = false,
+}: {
+  onDataViewCreated: (id: string, type: string, name?: string) => void;
+  allowAdHocDataView?: boolean;
+}) => {
+  const { dataViewEditor } = useMlKibana().services;
+  const canEditDataView = Boolean(dataViewEditor?.userPermissions.editDataView());
+  const dataViewEditorRef = useRef<() => void | undefined>();
+
+  const createNewDataView = useCallback(() => {
+    dataViewEditorRef.current = dataViewEditor?.openEditor({
```

Comment:
If this ref isn't going to be used to close the editor when navigating away, then I don't think we need the ref at all.
However, I think it would be good to use the ref as it was initially intended to close the editor when this button component is destroyed.

Reply:
Thanks for catching that, updated here [`2a655ac` (#166668)](https://github.com/elastic/kibana/pull/166668/commits/2a655ac8fca72ff954ce5caf9a69aa3922ecfce7) so it unmounts upon navigating away.